compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -53,7 +53,7 @@ pub fn finalize<'ll, 'tcx>(cx: &CodegenCx<'ll, 'tcx>) {
     let mut mapgen = CoverageMapGenerator::new();
 
     // Encode coverage mappings and generate function records
-    let mut function_data = Vec::new();
+    let mut function_data = Vec::with_capacity(function_coverage_map.len());
     for (instance, function_coverage) in function_coverage_map {
```

Comment:
Why not do a `let function_data: Vec<_> = function_coverage_map.into_iter().map(|(instance, function_coverage)| {...}).collect();`?

Reply:
This change shorter, and i don't think that converting `for` into iterator with closure in this place will give more readability or perf.